ejb3/src/main/java/org/jboss/as/ejb3/util/MdbValidationsUtil.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2011, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.ejb3.util;
+
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import org.jboss.as.ejb3.logging.EjbLogger;
+import org.jboss.as.server.deployment.DeploymentUnitProcessingException;
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.MethodInfo;
+
+/**
+ * @author Romain Pelisse - romain@redhat.com
+ */
+public final class MdbValidationsUtil {
+
+    private MdbValidationsUtil() {
+
+    }
+
+    /**
+     * Returns true if the passed <code>mdbClass</code> meets the requirements set by the EJB3 spec about bean implementation
+     * classes. The passed <code>mdbClass</code> must not be an interface and must be public and not final and not abstract. If
+     * it passes these requirements then this method returns true. Else it returns false.
+     *
+     * @param mdbClass The MDB class
+     * @return
+     * @throws DeploymentUnitProcessingException
+     */
+    public static Collection<MdbValidityStatus> assertMDBClassValidity(final ClassInfo mdbClass) throws DeploymentUnitProcessingException {
+        Collection<MdbValidityStatus> mdbComplianceIssueList = new ArrayList<>(MdbValidityStatus.values().length);
+        final String className = mdbClass.name().toString();
+        verifyModifiers(className, mdbClass.flags(), mdbComplianceIssueList);
+        for (MethodInfo method : mdbClass.methods()) {
+            final String methodName = method.name();
+            if ("onMessage".equals(methodName)) {
+                verifyOnMessageMethod(className, methodName, method.flags(), mdbComplianceIssueList);
```

Comment:
Although I am not a JMS expert, it seems that the onMessage method always has that name (or certainly that is the only thing which triggers the checks here), so I don't think you need the method name here and in the resulting log messages